frontend/src/component/common/RoleBadge/RoleBadge.tsx

```diff
@@ -6,12 +6,21 @@ import { RoleDescription } from 'component/common/RoleDescription/RoleDescriptio
 
 interface IRoleBadgeProps {
     roleId: number;
+    children?: string;
 }
 
-export const RoleBadge = ({ roleId }: IRoleBadgeProps) => {
+export const RoleBadge = ({ roleId, children }: IRoleBadgeProps) => {
     const { role } = useRole(roleId.toString());
 
-    if (!role) return null;
+    if (!role) {
+        if (children)
```

Comment:
Not sure I understand this. If there's no role, why would we show a successful badge?

If there's no role, it means we couldn't find the role, but in the [example below](https://github.com/Unleash/unleash/pull/4113/files#diff-0f1abeffe3343faae4ab07d78229d43fa1ec5d14dcedfefd76311dbc8c340058R144-R146) we're also trying to show the role name as children

Reply:
Previously this component took in a `roleId` as prop and used the `useRole` hook to fetch the role information. This worked great if you were Enterprise, not so much if you were not, since the GET role endpoint would not be available.

This PR adapts this component to have an optional graceful fallback in the children slot.

```typescript
if (!role) { // Confirms that we in fact could not fetch the role information based on the `roleId`, which means we're probably not Enterprise
  if (children) // There was a fallback children passed in
    return (
      <Badge color="success" icon={<UserIcon />}>
        {children} // Return a badge with the fallback info
      </Badge>
    );
return null; // Worst case scenario, do not render the component at all
}
```
If you have any ideas on how to implement this in a cleaner way please let me know 👍

Reply:
What if instead, you receive either a `roleId` or a role object (in the case the application already has the role)? I the UI has the role object from somewhere, you'd just provide it, otherwise you use the id to fetch the role (ofc when that endpoint is available)

Reply:
It's a bit more complicated than that because only the `useRole` endpoint returns `IRoleWithPermissions`, most of the time we're dealing with `IRole`. So if we want to provide detailed information it's better to accept a `roleId` and fetch the `IRoleWithPermissions` version of it.

We're basically saying the same thing but in a different order, where I prioritize `roleId` over `role`, and in this case my `role` is `children` which just includes the role name.

We could probably go through a heavier refactor here but maybe it belongs in a separate PR at a later point in time.